lib/base/log.c

```diff
@@ -199,38 +199,44 @@ open_syslog(heim_context context,
 }
 
 struct file_data {
-    const char *filename;
+    char *filename;
     const char *mode;
     struct timeval tv;
     FILE *fd;
     int disp;
 #define FILEDISP_KEEPOPEN       0x1
 #define FILEDISP_REOPEN         0x2
```

Comment:
It's not so much that we want to reopen on every call, but that we want to catch logfile rotations and reopen when logs rotate.  Except maybe for devices? we should just `stat()` the path once a second and re-open only when the file changes ID.